lib/transmission.js

```diff
@@ -47,101 +46,75 @@ var toApiFormat = function(input) {
   return model;
 };
 
-/**
- * Private Method for putting URL construction logic into one place
- *
- * @returns string Fully constructed URL for contacting the Transmissions API
- */
-var constructURL = function () {
-  return config.options.protocol + '://' + config.options.host + (config.options.port ? ':' + config.options.port: '') + '/api/' + config.options.version + '/transmissions';
-};
-
-/**
- *  Private Method for issuing GET request to Transmissions API
- *
- *  This method is responsible for getting the collection _and_
- *  a specific entity from the Transmissions API
- *
- *  If TransmissionID parameter is omitted, then we fetch the collection
- *
- *  @param TransmissionID string Transmission ID of specific Transmission to retrieve
- *  @param callback function
- */
-var fetch = function(transmissionID, callback) {
-  var cb = callback
-    , options = {
-      url: constructURL(),
-      headers: {
-        authorization: config.options.key
-      },
-      strictSSL: config.options.strictSSL
-    };
-
-  /*
-   * Check that someone passed in a real Transmission ID, if not,
-   * then the transmissionID parameter is acutally our callback
-   */
-  if (typeof transmissionID !== 'function' && typeof transmissionID !== undefined) {
-    options.url = options.url + '/' + transmissionID;
-  } else {
-    cb = transmissionID;
-  }
-
-  // Fetch, Fido, Fetch! (We may want to abstract the actual HTTP requests in a helper when we have more libs)
-  request.get(options, function(err, res, body) {
-    if (err) {
-      cb(new Error('Unable to contact Transmissions API: ' + err));
-    } else if (res.statusCode === 404) {
-      cb(new Error('The specified Transmission ID does not exist'));
-    } else if (res.statusCode !== 200) {
-      cb(new Error('Received bad response from Transmission API: ' + res.statusCode));
-    } else {
-      cb(null, body);
-    }
-  });
-};
-
 /*
  * "Class" declaration, Transmissions exposes three functions, one for sending a transmission,
  * another for getting a list of transmissions that have been sent, and another for getting
  * info about a specific transmission
  */
-module.exports = {
-  send: function(transmissionBody, callback) {
-    var mappedInput = toApiFormat(transmissionBody);
+module.exports = function(client) {
+
+  return {
+    send: function (transmissionBody, callback) {
+      var mappedInput = toApiFormat(transmissionBody);
 
-    var options = {
-      url: constructURL(),
-      json: mappedInput,
-      headers: {
-        authorization: config.options.key
-      },
-      strictSSL: config.options.strictSSL
-    };
+      var options = {
+        uri: api,
+        json: mappedInput
+      };
+
+      client.post(options, callback /*function (err, res, body) {
+        if (err) {
```

Comment:
Somewhere in this thread is a question about how we want to handle the server response going forward. I've always pushed it forward, but at least in transmissions, it's being wrapped. This was the easiest way to get the tests to pass. If we want to set a standards, I can do that.